src/server/game/Spells/Spell.cpp

```diff
@@ -3450,15 +3464,18 @@ template <class Container>
 void Spell::DoProcessTargetContainer(Container& targetContainer)
 {
     for (TargetInfoBase& target : targetContainer)
+    {
+        if (target.MissedTarget(this))
+            continue;
+
         target.PreprocessTarget(this);
 
-    for (uint8 i = 0; i < MAX_SPELL_EFFECTS; ++i)
-        for (TargetInfoBase& target : targetContainer)
+        for (uint8 i = 0; i < MAX_SPELL_EFFECTS; ++i)
```

Comment:
This change completely breaks spell effect processing order. EFFECT_0 must be applied on all targets, then EFFECT_1 on all targets then EFFECT_2. What you did here is do all effects on target 1 then all effects on target 2 etc